src/rules/table-column-count.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Rule to disallow data rows in a GitHub Flavored Markdown table from having more cells than the header row
+ * @author Sweta Tanwar (@SwetaTanwar)
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{ RuleOptions: []; }>} TableColumnCountRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {TableColumnCountRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description:
+				"Disallow data rows in a GitHub Flavored Markdown table from having more cells than the header row",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/table-column-count.md",
+		},
+
+		messages: {
+			inconsistentColumnCount:
+				"Table column count mismatch (Expected: {{expectedCells}}, Actual: {{actualCells}}), extra data starting here will be ignored.",
+		},
+	},
+
+	create(context) {
+		return {
+			table(node) {
+				if (!node.children || node.children.length < 1) {
+					return;
+				}
```

Comment:
```suggestion
				if (node.children.length < 1) {
					return;
				}
```

Can I ask why this line is needed? I can't think of any cases where `node.children.length < 1`.

Also, the `Table` node always has a `children` property as mentioned in its type definition. Can we remove the `!node.children` check?

![image](https://github.com/user-attachments/assets/a5e84ae4-73ff-47ab-9246-443d57678ac8)

![image](https://github.com/user-attachments/assets/630b953c-eea3-4d14-8e2c-dc86cc556d46)

Reply:
Got it, makes sense, applying your suggestion